modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1261,33 +1261,45 @@ def compute_part_size(indexer, remote_part, part_idx, axis):
 
     @classmethod
     @wait_computations_if_benchmark_mode
-    def binary_operation(cls, left, func, right):
-        """
-        Apply a function that requires two ``PandasDataframe`` objects.
+    def nary_operation(cls, left, func, right: list):
+        r"""
+        Apply a function that requires multiple ``PandasDataframe`` objects.
+
+        This method assumes that all the partitions of the dataframes in left
+        and right have the same dimensions. For each position i, j in each
+        dataframe's partitions, the result has a partition at (i, j) whose data
+        is func(let_partitions[i,j], \*each_right_partitions[i,j]).
 
         Parameters
         ----------
         left : np.ndarray
             The partitions of left ``PandasDataframe``.
         func : callable
             The function to apply.
-        right : np.ndarray
-            The partitions of right ``PandasDataframe``.
+        right : list of np.ndarray
+            The list of partitions of other ``PandasDataframe``.
 
         Returns
         -------
         np.ndarray
             A NumPy array with new partitions.
         """
-        [part.drain_call_queue() for part in right.flatten()]
+        [
+            part.drain_call_queue()
+            for right_partitions in right
+            for part in right_partitions.flatten()
+        ]
```

Comment:
I think we should have a way to execute this in parallel somehow... cc @prutskov

Reply:
Do we really want to add some new API to run `drain_call_queue` in parallel?

`part.drain_call_queue` just runs remote calls asynchronously and it probably does not take a lot of time.

Reply:
I was under false impression that `.drain_call_queue()` might wait on some intermediate results. I now see I was wrong.